src/Nethermind/Nethermind.Core.Test/Specs/ChainSpecStyle/ChainSpecBasedSpecProviderTests.cs

```diff
@@ -66,42 +66,81 @@ public void Mainnet_loads_properly()
             ChainSpecBasedSpecProvider provider = new ChainSpecBasedSpecProvider(chainSpec);
             MainNetSpecProvider mainnet = MainNetSpecProvider.Instance;
 
-            IReleaseSpec oldSpec = mainnet.GetSpec(7280000);
-            IReleaseSpec newSpec = provider.GetSpec(7280000);
-
-            PropertyInfo[] propertyInfos = typeof(IReleaseSpec).GetProperties(BindingFlags.Public | BindingFlags.Instance);
-            foreach (PropertyInfo propertyInfo in propertyInfos)
+            var blockNumbersToTest = new List<long>
             {
-                object a = propertyInfo.GetValue(oldSpec);
-                object b = propertyInfo.GetValue(newSpec);
-
-                Assert.AreEqual(a, b, propertyInfo.Name);
-            }
+                0,
+                1,
+                MainNetSpecProvider.HomesteadBlockNumber - 1,
+                MainNetSpecProvider.HomesteadBlockNumber,
+                MainNetSpecProvider.TangerineWhistleBlockNumber - 1,
+                MainNetSpecProvider.TangerineWhistleBlockNumber,
+                MainNetSpecProvider.SpuriousDragonBlockNumber - 1,
+                MainNetSpecProvider.SpuriousDragonBlockNumber,
+                MainNetSpecProvider.ByzantiumBlockNumber - 1,
+                MainNetSpecProvider.ByzantiumBlockNumber,
+                MainNetSpecProvider.ConstantinopleFixBlockNumber - 1,
+                MainNetSpecProvider.ConstantinopleFixBlockNumber,
+                MainNetSpecProvider.IstanbulBlockNumber - 1,
+                MainNetSpecProvider.IstanbulBlockNumber,
+                MainNetSpecProvider.MuirGlacierBlockNumber - 1,
+                MainNetSpecProvider.MuirGlacierBlockNumber,
+                100000000, // far in the future
+            };
+            
+            CompareSpecProperties(mainnet, provider, blockNumbersToTest);
             
             Assert.AreEqual(0000000, provider.GetSpec(4369999).DifficultyBombDelay);
             Assert.AreEqual(3000000, provider.GetSpec(4370000).DifficultyBombDelay);
             Assert.AreEqual(3000000, provider.GetSpec(7279999).DifficultyBombDelay);
             Assert.AreEqual(3000000, provider.GetSpec(7279999).DifficultyBombDelay);
             Assert.AreEqual(5000000, provider.GetSpec(7280000).DifficultyBombDelay);
             Assert.AreEqual(5000000, provider.GetSpec(9199999).DifficultyBombDelay);
-            Assert.AreEqual(9000000, provider.GetSpec(92000000).DifficultyBombDelay);
+            Assert.AreEqual(9000000, provider.GetSpec(9200000).DifficultyBombDelay);
         }
-        
+
+        private static void CompareSpecProperties(ISpecProvider oldSpec, ISpecProvider newSpec, IEnumerable<long> blockNumbers)
+        {
+            foreach (long blockNumber in blockNumbers)
+            {
+                PropertyInfo[] propertyInfos = typeof(IReleaseSpec).GetProperties(BindingFlags.Public | BindingFlags.Instance);
```

Comment:
You can sue oldSpec.Should().BeEquivalentTo(newSpec) no need to manually compare object structure and property values.

https://fluentassertions.com/objectgraphs/